src/services/aave.js

```diff
@@ -0,0 +1,244 @@
+    ));
+  }
+
+  async getAaveTokenAddress(assetAddress: string): Promise<?string> {
+    if (!this.aaveTokenAddresses[assetAddress]) {
+      const lendingPoolCoreContract = await this.getLendingPoolCoreContract();
+      if (!lendingPoolCoreContract) return null;
+      this.aaveTokenAddresses[assetAddress] = await lendingPoolCoreContract.getReserveATokenAddress(assetAddress);
+    }
+    return Promise.resolve(this.aaveTokenAddresses[assetAddress]);
+  }
+
+  async getAaveTokenContractForAsset(assetAddress: string): Promise<?Object> {
+    if (!this.lendingPoolAddressesProvider) return null;
+
+    const aaveTokenAddress = await this.getAaveTokenAddress(assetAddress);
+    if (!aaveTokenAddress) return null;
+
+    return Promise.resolve(getContract(
+      aaveTokenAddress,
+      AAVE_TOKEN_ABI,
+    ));
+  }
+
+  async getAaveTokenAddresses(): Promise<string[]> {
+    const lendingPoolCoreContract = await this.getLendingPoolCoreContract();
+    if (!lendingPoolCoreContract) return [];
+
+    const reserveAddresses = await lendingPoolCoreContract.getReserves().catch(() => []);
+
+    await Promise.all(reserveAddresses.map((address) => this.getAaveTokenAddress(address)));
+
+    return Promise.resolve((Object.values(this.aaveTokenAddresses): any));
+  }
+
+  async getSupportedDeposits(accountAssets: Asset[], supportedAssets: Asset[]): Promise<AssetToDeposit[]> {
+    const lendingPoolCoreContract = await this.getLendingPoolCoreContract();
+    if (!lendingPoolCoreContract) return Promise.resolve([]);
+
+    const poolAddresses = await lendingPoolCoreContract.getReserves().catch(() => []);
+
+    return poolAddresses.reduce((pool, reserveAddress) => {
+      const assetData = getAssetDataByAddress(accountAssets, supportedAssets, reserveAddress);
+      if (!isEmpty(assetData)) pool.push(assetData);
+      return pool;
+    }, []);
+  }
+
+  async getAssetsToDeposit(accountAssets: Asset[], supportedAssets: Asset[]): Promise<AssetToDeposit[]> {
+    const supportedDeposits = await this.getSupportedDeposits(accountAssets, supportedAssets);
+
+    const lendingPoolContract = await this.getLendingPoolContract();
+    if (!lendingPoolContract) return Promise.resolve([]);
+
+    return Promise.all(supportedDeposits.map(async (reserveAsset) => {
+      const reserveData = await lendingPoolContract
+        .getReserveData(reserveAsset.address)
+        .catch(() => ([]));
+      const earnInterestRate = rayToNumeric(reserveData[4]) * 100; // %
+      return {
+        ...reserveAsset,
+        earnInterestRate,
+      };
+    }));
+  }
+
+  async fetchAccountDepositedAsset(accountAddress: string, asset: Asset): Promise<$Shape<DepositedAsset>> {
+    const lendingPoolContract = await this.getLendingPoolContract();
+    if (!lendingPoolContract) return Promise.resolve({});
+
+    const depositedAssetData = await lendingPoolContract
+      .getUserReserveData(asset.address, accountAddress)
+      .catch(() => ([]));
```

Comment:
Same as above regarding `Sentry.captureException(err);` - at this point i just want to say - if _you_ know why we're not capturing errors here and that this is the correct solution then definitely please let me know :)